pages/_app.tsx

```diff
@@ -1,28 +1,69 @@
+import { useState, useEffect } from 'react'
 import type { AppProps } from 'next/app'
+import { useRouter } from 'next/router'
 import { Provider as ReduxProvider } from 'react-redux'
 import Head from 'next/head'
 import { SnackbarProvider } from 'notistack'
-
 import AuthWrapper from '@/components/auth'
 import { store } from '@/lib/redux/store'
-
+import BalloonLoader from '@/components/loader'
 import FormDrawer from '@/components/drawer'
+import styled from '@emotion/styled'
 import '@/styles/globals.css'
 
 export default function App({ Component, pageProps }: AppProps) {
+    const [loading, setLoading] = useState(true)
+    const router = useRouter()
+
+    const handleRouteChangeStart = () => setLoading(true)
+    const handleRouteChangeComplete = () => setLoading(false)
+
+    useEffect(() => {
+        const timer = setTimeout(() => setLoading(false), 1500)
```

Comment:
My only problem with this is, we are supposed to be querying most of the data (id imagine maybe we arent querying instagram until we hit that page, but idk i havent looked, but either way we should be querying alllll of the data at once) if not all of the data at once, when app first loads. I don't think it's necessary at all to display the loader every time we change routes. The loader should only display per page if that page doesnt have data ready to show